File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def isp(val):
            if val == val[::-1]:
                return True
            return False

        
        ans = ""
        for i in range(len(s)):
            l, r = i, i
            while l >= 0 and r < len(s) and isp(s[l:r+1]):
                if (r-l+1) > len(ans):
                    ans = s[l:r+1]
                    print(ans)
                l -= 1
                r += 1

            l, r = i, i+1
            while l >= 0 and r < len(s) and isp(s[l:r+1]):
                if (r-l+1) > len(ans):
                    print(ans)
                    ans = s[l:r+1]
                l -= 1
                r += 1
            
        return ans



        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        // We need to first make a prefix array, then a postifix array
        // The product of both these arrays form the answers 
        // THe prefix array is the product of all the numbers, not including it itself 

        int end = nums.length - 1;

        int [] ans = new int[nums.length]; 

        ans[0] = 1; 
        for (int i = 1; i < nums.length; i++) {
            // Prefix time
            ans[i] = ans[i-1] * nums[i-1]; 
        }
        // The end of the array is just the prefix till then 
        // So i dont need to put a 1 
        // Just continue calculating the postfix
        int postfix = 1; 
        for (int i = end; i >= 0; i--) {
            ans[i] = ans[i] * postfix; 
            postfix = nums[i] * postfix;

        }
        return ans;
    }
}  
